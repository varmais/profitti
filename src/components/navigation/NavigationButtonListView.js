import React, { Component, PropTypes } from 'react';
import { ListView, Text, View } from 'react-native';
import { navigatorPropTypes } from '../../helpers/PropTypes';
import Separator from '../common/Separator';
import AppStyles from '../../helpers/Styles';

export default class NavigationButtonListView extends Component {
  static propTypes = {
    items: PropTypes.array.isRequired,
    title: PropTypes.string.isRequired,
    navigation: navigatorPropTypes(),
    renderRow: PropTypes.func.isRequired
  };

  dataSource = new ListView.DataSource({
    rowHasChanged: (r1, r2) => r1.guid !== r2.guid
  });

  render () {
    const { items, title, renderRow } = this.props;
    return (
      <View style={AppStyles.container}>
        <View style={AppStyles.titleContainer}>
          <Text style={AppStyles.title}>
            {title}
          </Text>
        </View>
        <ListView
          dataSource={this.dataSource.cloneWithRows(items)}
          renderRow={renderRow}
          renderSeparator={() => <Separator />}
          removeClippedSubviews={false}
          {...this.props}
        />
      </View>
    );
  }
}
