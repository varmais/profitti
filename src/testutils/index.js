import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

global.React = React;
global.renderer = renderer;

export function createNavigationProp (params = {}) {
  return {
    state: {params: params},
    navigate: () => {}
  }
}