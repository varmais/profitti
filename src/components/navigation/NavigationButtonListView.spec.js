import {createNavigationProp} from '../../testutils';
import { View } from 'react-native';
import NavigationButtonListView from './NavigationButtonListView';

describe('components:navigation:NavigationButtonListView', () => {
  const renderRow = () => <View />;
  const items = [{item: 1}, {item: 4}];
  const title = 'title';

  test('renders NavigationButtonListView', () => {
    const snapshot = renderer.create(<NavigationButtonListView
      items={items}
      renderRow={renderRow}
      title={title}
      navigation={createNavigationProp()}
    />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
