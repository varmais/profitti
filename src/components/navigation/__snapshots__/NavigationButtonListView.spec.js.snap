// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:navigation:NavigationButtonListView renders NavigationButtonListView 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#caccce",
      "borderRadius": 4,
      "borderWidth": 1,
      "marginBottom": 8,
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#e6e6e6",
        "borderBottomWidth": 0.5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "flex-start",
          "color": "#222",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginVertical": 16,
        }
      }
    >
      title
    </Text>
  </View>
  <RCTScrollView
    dataSource={
      ListViewDataSource {
        "items": 2,
      }
    }
    items={
      Array [
        Object {
          "item": 1,
        },
        Object {
          "item": 4,
        },
      ]
    }
    navigation={
      Object {
        "navigate": [Function],
        "state": Object {
          "params": Object {},
        },
      }
    }
    removeClippedSubviews={false}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
    renderSeparator={[Function]}
    title="title"
  >
    <View>
      <View />
      <View />
    </View>
  </RCTScrollView>
</View>
`;
